Legacy formatting codes issue

### Skript/Server Version

```markdown
/sk info
[16:46:48 INFO]: [Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
[16:46:48 INFO]: [Skript] Skript's documentation can be found here: https://docs.skriptlang.org/
[16:46:48 INFO]: [Skript] Skript's tutorials can be found here: https://docs.skriptlang.org/tutorials
[16:46:48 INFO]: [Skript] Server Version: 1.21.11-131-6d5b910 (MC: 1.21.11)
[16:46:48 INFO]: [Skript] Skript Version: 2.15.2 (skriptlang-github)
[16:46:48 INFO]: [Skript] Installed Skript Addons: 
[16:46:48 INFO]: [Skript]  - SkBee v3.22.1 (https://github.com/ShaneBeee/SkBee)
[16:46:48 INFO]: [Skript]  - skript-worldguard v1.0.1 (https://github.com/SkriptLang/skript-worldguard)
[16:46:48 INFO]: [Skript] Installed dependencies: 
[16:46:48 INFO]: [Skript]  - WorldGuard v7.0.16+2355-f7fded2
```

### Bug Description

Returning tool of player, if said tool has any <hex> in its name or lore, it will run this error in the console and halt all code

### Expected Behavior

Return the tool of player or simply identify the item in a player's inventory
this line specificly is when I came across this,
if player has item with custom model data 3001:
any items such as:
emerald named "<gradient:#007c19:#80f3ab>Faction emerald" with custom model data 3001 with lore "<gray>Valuable" and "<gray>Abilities:" and "    <light blue>Required to create a faction" and  "   <light blue>Retains all faction progress" 
or 
copper golem statue named "&f&f<gradient:#ec6223:#fffe20><b>Trophy Golem" with custom model data 3002 with lore "<gray>Valuable" and "<gray>Abilities:" and "    <light blue>Can be used faction members" and  "    <light blue>to level up capitals"
would halt the code completely



### Steps to Reproduce

```
# RIVAL ITEMS
command /rivalItems:
	permission: op
	trigger:
		set {rivalItems::factionemerald} to emerald named "<gradient:#007c19:#80f3ab>Faction emerald" with custom model data 3001 with lore "<gray>Valuable" and "<gray>Abilities:" and "    <light blue>Required to create a faction" and  "   <light blue>Retains all faction progress" 
		set item model of {rivalItems::factionemerald} to "minecraft:faction_emerald"


```

These are the steps to get the *exact* item
```
command /checktool:
	permission: op
	trigger:
		send "&7[&1CheckTool&7]&f Name  &7%name of player's tool%"
		send "&7[&1CheckTool&7]&f Plain &7%plain player's tool%"
		send "&7[&1CheckTool&7]&f Lore &7%lore of player's tool%"
		send "&7[&1CheckTool&7]&f CMD &7%custom model data of player's tool%"
		send "&7[&1CheckTool&7]&f Model &7%item model of player's tool%"
		send "&7[&1CheckTool&7]&f Tool &7%player's tool%"
		send "&7[&1CheckTool&7]&f &aAll completed"
```
and here's what I used to pinpoint the issue

### Errors or Screenshots

<img width="854" height="480" alt="Image" src="https://github.com/user-attachments/assets/68d0566d-7e84-4312-b62b-5e2b78acbcab" />

<img width="854" height="480" alt="Image" src="https://github.com/user-attachments/assets/dee45b69-ba87-451f-944e-75c4e68ad140" />

[16:46:06 INFO]: STILLCAN issued server command: /checktool
[16:46:06 ERROR]: #!#! 
[16:46:06 ERROR]: #!#! [Skript] Severe Error:
[16:46:06 ERROR]: #!#! 
[16:46:06 ERROR]: #!#! An unexpected error occurred with Skript. This issue is likely not your fault.
[16:46:06 ERROR]: #!#! You are using some plugins that alter how Skript works (addons).
[16:46:06 ERROR]: #!#! Full list of addons:
[16:46:06 ERROR]: #!#! SkBee v3.22.1 (https://github.com/ShaneBeee/SkBee)
[16:46:06 ERROR]: #!#! skript-worldguard v1.0.1 (https://github.com/SkriptLang/skript-worldguard)
[16:46:06 ERROR]: #!#! We could not identify related addons, it might also be a Skript issue.
[16:46:06 ERROR]: #!#! Try temporarily removing the listed plugins one by one to identify the cause.
[16:46:06 ERROR]: #!#! If removing a plugin resolves the issue, please report the problem to the plugin developer.
[16:46:06 ERROR]: #!#! 
[16:46:06 ERROR]: #!#! Stack trace:
[16:46:06 ERROR]: #!#! Caused by: net.kyori.adventure.text.minimessage.internal.parser.ParsingExceptionImpl: Legacy formatting codes have been detected in a MiniMessage string - this is unsupported behaviour. Please refer to the Adventure documentation (https://docs.papermc.io/adventure/) for more information.
        <gray>[<dark_blue>CheckTool<gray>]<white> Tool <gray>emerald named §x<black><black><gray>§C<dark_blue><blue>F<#098523>a§x<dark_blue><dark_green><dark_gray>§D<dark_green>§Ec§x<dark_blue>§B<blue><gold><dark_aqua><dark_gray>t§x<dark_green><dark_purple><blue>§E<dark_red><dark_aqua>i§x<dark_green>§E§A<gray><dark_red>§Do§x<dark_aqua><gray>§A§F<dark_purple><dark_gray>n§x<dark_red><black>§B<dark_gray><gold><dark_green> §x<dark_red><blue>§C<black><gold>§Ce§x<dark_purple><dark_green>§C<blue><gray><gray>m§x<dark_purple>§B§D<dark_blue><dark_gray><dark_blue>e§x<gold><dark_purple>§D§A<dark_gray>§Cr§x<gold>§E§E<dark_green><blue><gold>a§x<gray><gray>§E§B§A<dark_blue>l§x<dark_gray><black>§F<dark_aqua>§A§Bd
                                                                                                 ^^
[16:46:06 ERROR]: #!#!     at net.kyori.adventure.text.minimessage.internal.parser.TokenParser.parseString(TokenParser.java:170)
[16:46:06 ERROR]: #!#!     at net.kyori.adventure.text.minimessage.internal.parser.TokenParser.resolvePreProcessTags(TokenParser.java:112)
[16:46:06 ERROR]: #!#!     at net.kyori.adventure.text.minimessage.MiniMessageParser.parseToTree(MiniMessageParser.java:195)
[16:46:06 ERROR]: #!#!     at net.kyori.adventure.text.minimessage.MiniMessageParser.parseFormat(MiniMessageParser.java:209)
[16:46:06 ERROR]: #!#!     at net.kyori.adventure.text.minimessage.MiniMessageImpl.deserialize(MiniMessageImpl.java:82)
[16:46:06 ERROR]: #!#!     at net.kyori.adventure.text.minimessage.MiniMessageImpl.deserialize(MiniMessageImpl.java:46)
[16:46:06 ERROR]: #!#!     at Skript-2.15.2 (3).jar//org.skriptlang.skript.bukkit.text.TextComponentParser.parse(TextComponentParser.java:439)
[16:46:06 ERROR]: #!#!     at Skript-2.15.2 (3).jar//org.skriptlang.skript.bukkit.text.TextComponentParser.parseSafe(TextComponentParser.java:425)
[16:46:06 ERROR]: #!#!     at Skript-2.15.2 (3).jar//org.skriptlang.skript.bukkit.text.TextModule.lambda$initSelf$0(TextModule.java:28)
[16:46:06 ERROR]: #!#!     at Skript-2.15.2 (3).jar//ch.njol.skript.lang.util.ConvertedExpression.lambda$new$1(ConvertedExpression.java:87)
[16:46:06 ERROR]: #!#!     at Skript-2.15.2 (3).jar//org.skriptlang.skript.lang.converter.Converters.convert(Converters.java:513)
[16:46:06 ERROR]: #!#!     at Skript-2.15.2 (3).jar//ch.njol.skript.lang.util.ConvertedExpression.getArray(ConvertedExpression.java:213)
[16:46:06 ERROR]: #!#!     at Skript-2.15.2 (3).jar//org.skriptlang.skript.bukkit.text.elements.effects.EffMessage.execute(EffMessage.java:63)
[16:46:06 ERROR]: #!#!     at Skript-2.15.2 (3).jar//ch.njol.skript.lang.Effect.run(Effect.java:42)
[16:46:06 ERROR]: #!#!     at Skript-2.15.2 (3).jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:40)
[16:46:06 ERROR]: #!#!     at Skript-2.15.2 (3).jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:67)
[16:46:06 ERROR]: #!#!     at Skript-2.15.2 (3).jar//ch.njol.skript.lang.Trigger.execute(Trigger.java:33)
[16:46:06 ERROR]: #!#!     at Skript-2.15.2 (3).jar//ch.njol.skript.command.ScriptCommand.execute2(ScriptCommand.java:353)
[16:46:06 ERROR]: #!#!     at Skript-2.15.2 (3).jar//ch.njol.skript.command.ScriptCommand.lambda$execute$0(ScriptCommand.java:312)
[16:46:06 ERROR]: #!#!     at Skript-2.15.2 (3).jar//ch.njol.skript.command.ScriptCommand.execute(ScriptCommand.java:323)
[16:46:06 ERROR]: #!#!     at Skript-2.15.2 (3).jar//ch.njol.skript.command.ScriptCommand.onCommand(ScriptCommand.java:258)
[16:46:06 ERROR]: #!#!     at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45)
[16:46:06 ERROR]: #!#!     at io.papermc.paper.command.brigadier.bukkit.BukkitCommandNode$BukkitBrigCommand.run(BukkitCommandNode.java:83)
[16:46:06 ERROR]: #!#!     at com.mojang.brigadier.context.ContextChain.runExecutable(ContextChain.java:73)
[16:46:06 ERROR]: #!#!     at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:30)
[16:46:06 ERROR]: #!#!     at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:13)
[16:46:06 ERROR]: #!#!     at net.minecraft.commands.execution.UnboundEntryAction.lambda$bind$0(UnboundEntryAction.java:8)
[16:46:06 ERROR]: #!#!     at net.minecraft.commands.execution.CommandQueueEntry.execute(CommandQueueEntry.java:5)
[16:46:06 ERROR]: #!#!     at net.minecraft.commands.execution.ExecutionContext.runCommandQueue(ExecutionContext.java:104)
[16:46:06 ERROR]: #!#!     at net.minecraft.commands.Commands.executeCommandInContext(Commands.java:469)
[16:46:06 ERROR]: #!#!     at net.minecraft.commands.Commands.performCommand(Commands.java:374)
[16:46:06 ERROR]: #!#!     at net.minecraft.commands.Commands.performCommand(Commands.java:362)
[16:46:06 ERROR]: #!#!     at net.minecraft.server.network.ServerGamePacketListenerImpl.performUnsignedChatCommand(ServerGamePacketListenerImpl.java:2375)
[16:46:06 ERROR]: #!#!     at net.minecraft.server.network.ServerGamePacketListenerImpl.lambda$handleChatCommand$12(ServerGamePacketListenerImpl.java:2348)
[16:46:06 ERROR]: #!#!     at net.minecraft.server.TickTask.run(TickTask.java:18)
[16:46:06 ERROR]: #!#!     at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:177)
[16:46:06 ERROR]: #!#!     at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24)
[16:46:06 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1525)
[16:46:06 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:188)
[16:46:06 ERROR]: #!#!     at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:151)
[16:46:06 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1505)
[16:46:06 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1499)
[16:46:06 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.recordTaskExecutionTimeWhileWaiting(MinecraftServer.java:1225)
[16:46:06 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1341)
[16:46:06 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:388)
[16:46:06 ERROR]: #!#!     at java.base/java.lang.Thread.run(Thread.java:1583)
[16:46:06 ERROR]: #!#! 
[16:46:06 ERROR]: #!#! Skript: 2.15.2 (latest)
[16:46:06 ERROR]: #!#!     Flavor: skriptlang-github
[16:46:06 ERROR]: #!#!     Date: 23:29:38.148343771
[16:46:06 ERROR]: #!#! Bukkit: 1.21.11-R0.1-SNAPSHOT
[16:46:06 ERROR]: #!#! Minecraft: 1.21.11
[16:46:06 ERROR]: #!#! Java: 21.0.10 (OpenJDK 64-Bit Server VM 21.0.10+7-LTS)
[16:46:06 ERROR]: #!#! OS: Linux amd64 6.8.0-31-generic
[16:46:06 ERROR]: #!#! 
[16:46:06 ERROR]: #!#! Server platform: Paper
[16:46:06 ERROR]: #!#! 
[16:46:06 ERROR]: #!#! Current node: null
[16:46:06 ERROR]: #!#! Current item: message ("[CheckTool] Tool %tool of the player%" >> ch.njol.skript.lang.util.ConvertedExpression$$Lambda/0x00007680b1b01310@1820d87f: ConverterInfo{from=class java.lang.String,to=interface net.kyori.adventure.text.Component,converter=org.skriptlang.skript.bukkit.text.TextModule$$Lambda/0x00007680b1682628@13e98c90,flags=0}) to event-command sender
[16:46:06 ERROR]: #!#! Current trigger: command /checktool (simple event) (funcommands.sk, line 120)
[16:46:06 ERROR]: #!#! Thread: Server thread
[16:46:06 ERROR]: #!#! Language: english
[16:46:06 ERROR]: #!#! Link parse mode: DISABLED
[16:46:06 ERROR]: #!#! End of Error.
[16:46:06 ERROR]: #!#! 



### Other

first report kinda nervous 😳 


### Agreement

- [x] I have read the guidelines above and affirm I am following them with this report.

--- a/src/main/java/ch/njol/skript/aliases/ItemData.java
+++ b/src/main/java/ch/njol/skript/aliases/ItemData.java
@@ -23,6 +23,7 @@ import org.bukkit.inventory.meta.MusicInstrumentMeta;
 import org.bukkit.inventory.meta.PotionMeta;
 import org.bukkit.potion.PotionEffect;
 import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.text.TextComponentParser;
 
 import java.io.IOException;
 import java.io.NotSerializableException;
@@ -242,8 +243,10 @@ public class ItemData implements Cloneable, YggdrasilExtendedSerializable {
 		StringBuilder builder = new StringBuilder(Aliases.getMaterialName(this, plural));
 		ItemMeta meta = stack != null ? stack.getItemMeta() : null;
 		if (meta != null && meta.hasDisplayName()) {
-			builder.append(" ").append(m_named).append(" ");
-			builder.append(meta.getDisplayName());
+			builder.append(" ")
+				.append(m_named)
+				.append(" ")
+				.append(TextComponentParser.instance().toString(meta.displayName()));
 		}
 		return builder.toString();
 	}

